pkg/cosmos/client/gas_price_estimator.go

```diff
@@ -17,16 +19,77 @@ var _ GasPricesEstimator = (*FixedGasPriceEstimator)(nil)
 
 type FixedGasPriceEstimator struct {
 	gasPrices map[string]sdk.DecCoin
+	lggr      logger.SugaredLogger
 }
 
-func NewFixedGasPriceEstimator(prices map[string]sdk.DecCoin) *FixedGasPriceEstimator {
-	return &FixedGasPriceEstimator{gasPrices: prices}
+func NewFixedGasPriceEstimator(prices map[string]sdk.DecCoin, lggr logger.SugaredLogger) *FixedGasPriceEstimator {
+	return &FixedGasPriceEstimator{gasPrices: prices, lggr: lggr}
 }
 
 func (gpe *FixedGasPriceEstimator) GasPrices() (map[string]sdk.DecCoin, error) {
 	return gpe.gasPrices, nil
 }
 
+func (gpe *FixedGasPriceEstimator) GasPrice(coin string) sdk.DecCoin {
+	return gpe.gasPrices[coin]
+}
+
+func (gpe *FixedGasPriceEstimator) SetGasPrice(coin string, price sdk.DecCoin) {
```

Comment:
> same for SetGasPrice, does this/will this get called anywhere besides the test?
For SetGasPrice, this would be called to set the price of gas after it is being calculated. Initially i wanted to combine the calculate gas price function and combine it with set gas price function to have a combined function, but felt it was cleaner to have them seperated.

> should it test that the coin is already one of the existing coins provided in the initial gas price map?
Thanks for suggesting this error case. Will implement it, apologies for missing it.